cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/common/BatchFileFilter.java

```diff
@@ -80,7 +80,8 @@ public boolean accept(Path path) {
 
     //filter by file name using regex from configuration
     if (!useTimeFilter) {
-      Matcher matcher = regex.matcher(filePathName);
```

Comment:
Its a regex filter on the ``fileName``. The directory is passed separately. @sreevatsanraman that's how its supposed to work, right?